searchcore/src/vespa/searchcore/proton/summaryengine/summaryengine.cpp

```diff
@@ -114,10 +135,21 @@ SummaryEngine::getDocsums(DocsumRequest::UP req)
                 reply = snapshot->get()->getDocsums(*req); // use the first handler
             }
         }
+        updateDocsumMetrics(req->getTimeUsed().sec(), reply->docsums.size());
     }
     reply->request = std::move(req);
+
     return reply;
 }
 
+void
+SummaryEngine::updateDocsumMetrics(double latency_s, uint32_t numDocs)
+{
+    std::lock_guard guard(_lock);
+    DocsumMetrics & m = static_cast<DocsumMetrics &>(*_metrics);
```

Comment:
do we really need to typecast our own metrics?

Reply:
Typecasting makes even less sense now that the definition is in the header file.